src/wagtail_vector_index/backends/pgvector/backend.py

```diff
@@ -79,7 +79,7 @@ def similarity_search(
         ):
             embedding = pgvector_embedding.embedding
             doc = embedding.to_document()
-            yield SearchResponseDocument(id=doc.id, metadata=doc.metadata)
+            yield SearchResponseDocument(id=doc.embedding_pk, metadata=doc.metadata)
```

Comment:
**Question:** Shouldn't this be `SearchResponseDocument(embedding_pk=doc.embedding_pk, ...)` instead? What's the `id` here of?

Reply:
I don't even know why we're returning a `SearchResponseDocument` here rather than a `Document` which would make more sense.

I've restructured this so `backends` now return Documents.